tracer/src/Datadog.Trace/Span.cs

```diff
@@ -124,6 +125,8 @@ internal string ServiceName
 
         internal SpanContext Context { get; }
 
+        internal List<SpanLink> SpanLinks { get; set; }
```

Comment:
We probably want a private setter here so only we can initialize the list when it's null in `AddSpanLink()` and not allow other code to replace the list with a new one.

```suggestion
        internal List<SpanLink> SpanLinks { get; private set; }
```